### DISCUSSION TITLE
Add break button to timer

### DISCUSSION THREAD

**Discussion by A11yship** [neuefische/questions#101] [Web] [answered]
Hello,

I wrote a timer for my App and it works fine. Now i try to implement a break function. If the user clicks on the break button ist should pause the timer and clicking it a second timer should start the timer again with the remaining time.
At the moment I can start the timer, but stopping it has no effect on the timer. Clicking a third time on the Start/Break-Button starts a second timer parallel to the first timer.
I think there is a mistake with calling `clearInterval()` in my code. Maybe it is also a bad idea to use `useEffect` in this case.
I create a [PR with the current code base](https://github.com/A11yship/capstone-project/pull/17) and hier ist the [preview on vercel](https://capstone-project-git-digitaltimer-a11yship.vercel.app/). 
Thanks for your answers.

**Comment by mpagels** [comment, marked as answer]
Checkout this `codesandbox` I found.
https://codesandbox.io/s/timer-react-useeffect-d59gr?file=/src/App.js
and adapt it to your project. ;-)

Only if you get stuck, check this out:

<details>
  <summary>CurrentTask.js</summary>
  
```js

import {useEffect, useRef, useState} from 'react';

import useStore from '../../hooks/useStore';
import Button from '../Button/Button';

import StyledCurrentTask from './StyledCurrentTask';

export default function CurrentTask() {
	const tasks = useStore(state => state.tasks);
	const completeTask = useStore(state => state.deleteTask);

	const [min, setMin] = useState(tasks[0].time * 60);
	const [isOn, setIsOn] = useState(false);

	useEffect(() => {
		let interval;

		if (isOn) {
			interval = setInterval(() => setMin(min => min - 1), 1000);
		}
		return () => clearInterval(interval);
	}, [isOn]);

	// 👇️ get number of full minutes
	const minutes = Math.floor(min / 60);

	// 👇️ get remainder of seconds
	const seconds = min % 60;

	function padTo2Digits(num) {
		return num.toString().padStart(2, '0');
	}

	return (
		<StyledCurrentTask>
			{tasks.length ? (
				<>
					<p>{tasks[0].name}</p>
					<p>
						{padTo2Digits(minutes)}:{padTo2Digits(seconds)} min
					</p>

					<Button onClick={() => setIsOn(!isOn)}>{isOn ? 'Pause' : 'Start'}</Button>
					<Button onClick={() => completeTask(tasks[0].id)}>done</Button>
				</>
			) : (
				<p>Keine Aufgabe</p>
			)}
		</StyledCurrentTask>
	);
}
```
</details>

**Reply by A11yship** [reply]
Thank you. That helped me and I can solve the problem.
